src/SelfServe/SelfServeTypes.ts

```diff
@@ -59,16 +70,48 @@ export interface SelfServeDescriptor {
   refreshParams?: RefreshParams;
 }
 
+/**@internal */
 export type AnyDisplay = NumberInput | BooleanInput | StringInput | ChoiceInput | DescriptionDisplay;
 
-export abstract class SelfServeBaseClass {
-  public abstract initialize: () => Promise<Map<string, SmartUiInput>>;
-  public abstract onSave: (
+/**@internal */
+export type InputTypeValue = "number" | "string" | "boolean" | "object";
+
+export type initializeCallback =
+  /**
+   * @returns Promise of Map of propertyName => {@linkcode SmartUiInput} which will become the current state of the UI.
+   */
+  () => Promise<Map<string, SmartUiInput>>;
+
+export type onSaveCallback =
+  /**
+   * @param currentValues - The map of propertyName => {@linkcode SmartUiInput} corresponding to the current state of the UI
+   * @param baselineValues - The map of propertyName => {@linkcode SmartUiInput} corresponding to the initial state of the UI
+   */
+  (
     currentValues: Map<string, SmartUiInput>,
     baselineValues: ReadonlyMap<string, SmartUiInput>
   ) => Promise<OnSaveResult>;
+
+export abstract class SelfServeBaseClass {
```

Comment:
>SelfServeBaseClass [](http://example.com/codeflow?start=22&length=18)

please add a one liner; all self serve feature classes need to derive from this base class.